contract/src/interfaces/Types.sol

```diff
@@ -11,39 +11,53 @@ struct Signature {
     bytes32 s;
 }
 
+struct Amount {
+  address token; // ERC-20 token contract address
+  uint256 amount; // Token amount allocated
+}
+
 struct Allocation {
     address destination; // Where funds are sent on channel closure
     address token; // ERC-20 token contract address
     uint256 amount; // Token amount allocated
+    //TODO: consider using Amount
 }
 
 struct Channel {
-    address[2] participants; // List of participants in the channel [Host, Guest]
+    address[] participants; // List of participants in the channel [Host, Guest]
     address adjudicator; // Address of the contract that validates final states
     uint64 challenge; // Duration in second, Participants can dispute by submitting newer valid state during challenge
     uint64 nonce; // Unique per channel with same participants and adjudicator
 }
 
 struct State {
     bytes data; // Application data encoded, decoded by the adjudicator for business logic
-    Allocation[2] allocations; // Combined asset allocation and destination for each participant
+    Allocation[] allocations; // Combined asset allocation and destination for each participant
     Signature[] sigs; // stateHash signatures
 }
 
 enum Status {
-    VOID, // Channel was never active (zero-initialized)
-    PARTIAL, // Partial funding waiting for other participants
+    VOID, // VOID Channel was not created
+    INITIAL, // Channel is created and in funding process
     ACTIVE, // Channel fully funded and valid state
+    DISPUTE,
     FINAL, // This is the FINAL state, channel can be closed
-    INVALID // Channel state is invalid
-
 }
 
-// This struct has been moved to Custody.sol with additional fields
-// Kept here for backward compatibility, but should be migrated to use the Custody.sol version
+constant uint256 CREATOR = 0; // participant index for the channel creator
+constant uint256 BROKER = 1; // participant index for the broker
+
+// Funding protocol use State with expected deposits
+constant uint32 CHANOPEN = 7877; // State.data value for funding stateHash
+constant uint32 CHANCLOSE = 7879; // State.data value for closing stateHash
```

Comment:
In future we might want to move `open` and `close` constants out of IChannel to allow each Adjudicator to define its own values. For example, some may allow start or final states not only contain a magic number, but rather be meaningful for their respective protocol. This way we save 1 participants interaction in each open and close.

Reply:
Dont speak to me anymore about custom funding protocol, I dont want to hear it ever again